core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -351,7 +354,35 @@ public void invalidateTable(SessionContext context, TableIdentifier ident) {}
   @Override
   public Table registerTable(
       SessionContext context, TableIdentifier ident, String metadataFileLocation) {
-    throw new UnsupportedOperationException("Register table is not supported");
+    Preconditions.checkArgument(
+        ident != null && isValidIdentifier(ident), "Invalid identifier: %s", ident);
+    Preconditions.checkArgument(
+        metadataFileLocation != null && !metadataFileLocation.isEmpty(),
+        "Cannot register an empty metadata file location as a table");
+
+    // Throw an exception if this table already exists in the catalog.
+    if (tableExists(context, ident)) {
+      throw new AlreadyExistsException("Table already exists: %s", ident);
+    }
+
+    TableMetadata metadata = TableMetadataParser.read(io, metadataFileLocation);
+    Map<String, String> tableProperties =
+        ImmutableMap.<String, String>builder()
+            .putAll(metadata.properties())
+            .put(BaseMetastoreTableOperations.METADATA_LOCATION_PROP, metadataFileLocation)
+            .build();
+
+    return buildTable(context, ident, metadata.schema())
+        .withLocation(metadata.location())
+        .withProperties(tableProperties)
+        .withPartitionSpec(metadata.spec())
+        .withSortOrder(metadata.sortOrder())
+        .create();
+  }
+
+  private boolean isValidIdentifier(TableIdentifier tableIdentifier) {
```

Comment:
do we need this, given that it always returns true? I know we do the same in `BaseMetastoreCatalog` where catalog implementations can override this behavior but I don't think we need this here in particular. 
You might rather want to call `checkIdentifierIsValid(..)` in this case.

Reply:
Yes, I have copied `isValidIdentifier` from `BaseMetastoreCatalog` implementation. Thanks for the input. I will use `checkIdentifierIsValid` over here.